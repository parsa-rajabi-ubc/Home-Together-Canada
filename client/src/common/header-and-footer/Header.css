.nav-icon {
    @apply inline-block px-2 py-2 mr-3 transition duration-200 ease-in-out border-transparent rounded-md;
}

.nav-icon:hover {
    @apply bg-white text-black opacity-50;
}

.nav-btn {
    @apply items-center justify-center w-full px-6 py-2 mr-3 transition duration-200 ease-in-out bg-white border-transparent rounded-md opacity-75;
}

.nav-btn:hover {
    @apply bg-orange-400;
}