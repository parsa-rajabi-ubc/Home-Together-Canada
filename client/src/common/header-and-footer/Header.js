/**
 * @Author:     Parsa Rajabi
 * @Created:    2020.11.09
 *
 * @Description: header for the website which contains links to Connect with Members, Services, Classifieds, About Us and Account Page
 *
 */

import React from 'react';
import '../../tailwind.output.css';
import {Link} from "react-router-dom";
import LoginService from '../../services/LoginService';
import Button from "../forms/Button";

const Header = () => {

    const logout = () => {
        LoginService.logoutUser()
            .then(res => res.json())
            .then(() => {
                alert('You have been logged out.')
            })
    }


    // TODO: Dynamically generate buttons in header based on authentication and user privileges

    return (
        <div>
            {/*TODO: change bg color to igreen*/}
            <nav className="top-0 flex w-full bg-green-400 tex-black">
                <div className="container flex items-center justify-between w-full py-2 mx-auto mt-0 whitespace-no-wrap">
                    <div className="flex flex-no-wrap items-center ml-4">
                        <Link to={'/'}
                              className="text-lg font-bold lg:text-xl">
                            Home Together Canada
                        </Link>
                    </div>

                    {/*TODO: move classNames to header.css */}
                    {/* Middle of Nav */}
                    <div
                        className="flex-no-wrap hidden w-full p-4 mt-2 tex-black lg:flex lg:items-center lg:w-auto lg:block lg:mt-0 lg:bg-transparent lg:p-0">
                        <div className="items-center justify-end flex-1 mr-16 list-reset lg:flex">
                            <Link to={'/members'}
                                  className="inline-block px-4 py-2 mr-3 transition duration-200 ease-in-out border-transparent rounded-md hover:bg-white hover:text-black hover:opacity-50">
                                Connect with Members
                            </Link>
                            <Link to={'/services'}
                                  className="inline-block px-4 py-2 mr-3 transition duration-200 ease-in-out border-transparent rounded-md hover:bg-white hover:text-black hover:opacity-50">
                                Services
                            </Link>
                            <Link to={'/classifieds'}
                                  className="inline-block px-4 py-2 mr-3 transition duration-200 ease-in-out border-transparent rounded-md hover:bg-white hover:text-black hover:opacity-50">
                                Classifieds
                            </Link>
                            <Link to={'/about'}
                                  className="inline-block px-4 py-2 mr-3 transition duration-200 ease-in-out border-transparent rounded-md hover:bg-white hover:text-black hover:opacity-50">
                                About Us
                            </Link>
                            <Link to={'/faq'}
                                  className="inline-block px-4 py-2 mr-3 no-underline transition duration-200 ease-in-out border-transparent rounded-md hover:bg-white hover:text-black hover:opacity-50">
                                FAQ
                            </Link>
                        </div>

                        {/* Login and Sign Up Buttons */}
                        <Link to={'/login'}
                              className="items-center justify-center w-full px-6 py-2 mr-4 transition duration-200 ease-in-out bg-white border-transparent rounded-md opacity-75 hover:bg-orange-400">Login
                        </Link>
                        <Link to={'/registration'}
                              className="items-center justify-center w-full px-4 py-2 mr-4 transition duration-200 ease-in-out bg-white border-transparent rounded-md opacity-75 hover:bg-orange-400">Sign
                            Up
                        </Link>
                        <Button
                            className="outline-none	items-center justify-center w-full px-4 py-2 transition duration-200 ease-in-out bg-white border-transparent rounded-md opacity-75 cursor-pointer hover:bg-orange-400"
                            value={'Logout'}
                            onClick={logout}/>
                    </div>
                </div>
                {/*Bottom Border*/}
                <hr className="py-0 my-0 border-b border-gray-200 opacity-25"/>
            </nav>
        </div>
    )
}

export default Header;