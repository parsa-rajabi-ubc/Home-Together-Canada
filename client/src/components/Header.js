// Component Description:
// This component is the header for the website which contains links to Connect with Members, Services, Classifieds, About Us and Account Page

import React from 'react';
import '../tailwind.output.css';

const Header = () => {
    return (
        <div>
            <div className="realtive">
                <div className="">
                    <div className="flex justify-between items-center">
                        <div>
                            <a href="#" className="flex">
                                Home Together Canada
                            </a>
                        </div>
                        <nav className="md:flex space-x-10">
                            <a href="#">
                                Connect with Members
                            </a>
                            <a href="#">
                                Services
                            </a>
                            <a href="#">
                                Classifieds
                            </a>
                            <a href="#">
                                About Us
                            </a>
                            <a href="#">
                                FAQ
                            </a>
                        </nav>

                        <div>
                            {/*TODO: Add href to login page*/}
                            <button href="#"
                                    className="border">
                                Login
                            </button>
                            {/*TODO: Add href to registration page*/}
                            <button href="#"
                                    className="border">
                                Sign Up
                            </button>
                        </div>
                    </div>
                </div>
            </div>
        </div>
    )
}

export default Header;