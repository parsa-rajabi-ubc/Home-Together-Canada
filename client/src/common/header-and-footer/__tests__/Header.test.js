import React from 'react';
import renderer from  'react-test-renderer'
import Header from "../Header";
import {BrowserRouter, Link} from "react-router-dom";

jest.mock('react-redux', () => ({
    connect: () => {
        return (component) => {
            return component
        };
    }
}));

const setAccountType = jest.fn();
const setIsAdmin = jest.fn();
const setAuthenticated = jest.fn();
const setActive = jest.fn();

describe('Header', () => {
    describe('Header structure', () => {
        it('should match snapshot test', () => {
            // given
            const props = {
                setAccountType,
                setIsAdmin,
                setAuthenticated,
                setActive,
                isAdmin: false,
                accountType: null,
                authenticated: false
            };
            //when
            const component = renderer.create(<BrowserRouter><Header {...props}/></BrowserRouter>);
            const tree = component.toJSON();
            //then
            expect(tree).toMatchSnapshot();
        });
    });

    describe('Search Member Profiles button', () => {
        it('should show Search Member Profiles button when the user is unregistered', () => {
            // given
            const props = {
                setAccountType,
                setIsAdmin,
                setAuthenticated,
                setActive,
                isAdmin: false,
                accountType: null,
                authenticated: false
            };

            // when
            const testInstance = renderer.create(<BrowserRouter><Header {...props}/></BrowserRouter>).root;
            const connectWithMembersButton = testInstance.findAllByType(Link)
                .find(element => element.props.children === 'Search Member Profiles');

            // then
            expect(connectWithMembersButton).toBeDefined();
        });
        it('should show Search Member Profiles button when the user is a member', () => {
            // given
            const props = {
                setAccountType,
                setIsAdmin,
                setAuthenticated,
                setActive,
                isAdmin: false,
                accountType: 'member',
                authenticated: true
            };

            // when
            const testInstance = renderer.create(<BrowserRouter><Header {...props}/></BrowserRouter>).root;
            const connectWithMembersButton = testInstance.findAllByType(Link)
                .find(element => element.props.children === 'Search Member Profiles');

            // then
            expect(connectWithMembersButton).toBeDefined();
        });
        it('should not show Search Member Profiles button when the user is a business', () => {
            // given
            const props = {
                setAccountType,
                setIsAdmin,
                setAuthenticated,
                setActive,
                isAdmin: false,
                accountType: 'business',
                authenticated: true
            };

            // when
            const testInstance = renderer.create(<BrowserRouter><Header {...props}/></BrowserRouter>).root;
            const connectWithMembersButton = testInstance.findAllByType(Link)
                .find(element => element.props.children === 'Search Member Profiles');

            // then
            expect(connectWithMembersButton).toBeUndefined();
        });
    });

    describe('Home Sharing Services button', () => {
        it('should show Home Sharing Services button when the user is unregistered', () => {
            // given
            const props = {
                setAccountType,
                setIsAdmin,
                setAuthenticated,
                setActive,
                isAdmin: false,
                accountType: null,
                authenticated: false
            };

            // when
            const testInstance = renderer.create(<BrowserRouter><Header {...props}/></BrowserRouter>).root;
            const servicesButton = testInstance.findAllByType(Link)
                .find(element => element.props.children === 'Home Sharing Services');

            // then
            expect(servicesButton).toBeDefined();
        });
        it('should show Home Sharing Services button when the user is a member', () => {
            // given
            const props = {
                setAccountType,
                setIsAdmin,
                setAuthenticated,
                setActive,
                isAdmin: false,
                accountType: 'member',
                authenticated: true
            };

            // when
            const testInstance = renderer.create(<BrowserRouter><Header {...props}/></BrowserRouter>).root;
            const servicesButton = testInstance.findAllByType(Link)
                .find(element => element.props.children === 'Home Sharing Services');

            // then
            expect(servicesButton).toBeDefined();
        });
        it('should show Home Sharing Services button when the user is a business', () => {
            // given
            const props = {
                setAccountType,
                setIsAdmin,
                setAuthenticated,
                setActive,
                isAdmin: false,
                accountType: 'business',
                authenticated: true
            };

            // when
            const testInstance = renderer.create(<BrowserRouter><Header {...props}/></BrowserRouter>).root;
            const servicesButton = testInstance.findAllByType(Link)
                .find(element => element.props.children === 'Home Sharing Services');

            expect(servicesButton).toBeDefined();
        });
    });

    describe('Local Classifieds button', () => {
        it('should show Local Classifieds button when the user is unregistered', () => {
            // given
            const props = {
                setAccountType,
                setIsAdmin,
                setAuthenticated,
                setActive,
                isAdmin: false,
                accountType: null,
                authenticated: false
            };

            // when
            const testInstance = renderer.create(<BrowserRouter><Header {...props}/></BrowserRouter>).root;
            const classifiedsButton = testInstance.findAllByType(Link)
                .find(element => element.props.children === 'Local Classifieds');

            // then
            expect(classifiedsButton).toBeDefined();
        });
        it('should show Local Classifieds button when the user is a member', () => {
            // given
            const props = {
                setAccountType,
                setIsAdmin,
                setAuthenticated,
                setActive,
                isAdmin: false,
                accountType: 'member',
                authenticated: true
            };

            // when
            const testInstance = renderer.create(<BrowserRouter><Header {...props}/></BrowserRouter>).root;
            const classifiedsButton = testInstance.findAllByType(Link)
                .find(element => element.props.children === 'Local Classifieds');

            // then
            expect(classifiedsButton).toBeDefined();
        });
        it('should show Local Classifieds button when the user is a business', () => {
            // given
            const props = {
                setAccountType,
                setIsAdmin,
                setAuthenticated,
                setActive,
                isAdmin: false,
                accountType: 'business',
                authenticated: true
            };

            // when
            const testInstance = renderer.create(<BrowserRouter><Header {...props}/></BrowserRouter>).root;
            const classifiedsButton = testInstance.findAllByType(Link)
                .find(element => element.props.children === 'Local Classifieds');

            // then
            expect(classifiedsButton).toBeDefined();
        });
    });

    describe('About Us button', () => {
        it('should show About Us button when the user is unregistered', () => {
            // given
            const props = {
                setAccountType,
                setIsAdmin,
                setAuthenticated,
                setActive,
                isAdmin: false,
                accountType: null,
                authenticated: false
            };

            // when
            const testInstance = renderer.create(<BrowserRouter><Header {...props}/></BrowserRouter>).root;
            const aboutUsButton = testInstance.findAllByType(Link)
                .find(element => element.props.children === 'About Us');

            // then
            expect(aboutUsButton).toBeDefined();
        });
        it('should show About Us button when the user is a member', () => {
            // given
            const props = {
                setAccountType,
                setIsAdmin,
                setAuthenticated,
                setActive,
                isAdmin: false,
                accountType: 'member',
                authenticated: true
            };

            // when
            const testInstance = renderer.create(<BrowserRouter><Header {...props}/></BrowserRouter>).root;
            const aboutUsButton = testInstance.findAllByType(Link)
                .find(element => element.props.children === 'About Us');

            // then
            expect(aboutUsButton).toBeDefined();
        });
        it('should show About Us button when the user is a business', () => {
            // given
            const props = {
                setAccountType,
                setIsAdmin,
                setAuthenticated,
                setActive,
                isAdmin: false,
                accountType: 'business',
                authenticated: true
            };

            // when
            const testInstance = renderer.create(<BrowserRouter><Header {...props}/></BrowserRouter>).root;
            const aboutUsButton = testInstance.findAllByType(Link)
                .find(element => element.props.children === 'About Us');

            // then
            expect(aboutUsButton).toBeDefined();
        });
    });

    describe('FAQ button', () => {
        it('should show FAQ button when the user is unregistered', () => {
            // given
            const props = {
                setAccountType,
                setIsAdmin,
                setAuthenticated,
                setActive,
                isAdmin: false,
                accountType: null,
                authenticated: false
            };

            // when
            const testInstance = renderer.create(<BrowserRouter><Header {...props}/></BrowserRouter>).root;
            const faqButton = testInstance.findAllByType(Link)
                .find(element => element.props.children === 'FAQ');

            // then
            expect(faqButton).toBeDefined();
        });
        it('should show FAQ button when the user is a member', () => {
            // given
            const props = {
                setAccountType,
                setIsAdmin,
                setAuthenticated,
                setActive,
                isAdmin: false,
                accountType: 'member',
                authenticated: true
            };

            // when
            const testInstance = renderer.create(<BrowserRouter><Header {...props}/></BrowserRouter>).root;
            const faqButton = testInstance.findAllByType(Link)
                .find(element => element.props.children === 'FAQ');

            // then
            expect(faqButton).toBeDefined();
        });
        it('should show FAQ button when the user is a business', () => {
            // given
            const props = {
                setAccountType,
                setIsAdmin,
                setAuthenticated,
                setActive,
                isAdmin: false,
                accountType: 'business',
                authenticated: true
            };

            // when
            const testInstance = renderer.create(<BrowserRouter><Header {...props}/></BrowserRouter>).root;
            const faqButton = testInstance.findAllByType(Link)
                .find(element => element.props.children === 'FAQ');

            // then
            expect(faqButton).toBeDefined();
        });
    });

    describe('Create Listing button', () => {
        it('should show not show Create Listing button when the user is unregistered', () => {
            // given
            const props = {
                setAccountType,
                setIsAdmin,
                setAuthenticated,
                setActive,
                isAdmin: false,
                accountType: null,
                authenticated: false
            };

            // when
            const testInstance = renderer.create(<BrowserRouter><Header {...props}/></BrowserRouter>).root;
            const createListingButton = testInstance.findAllByType(Link)
                .find(element => element.props.children === 'Create Listing');

            // then
            expect(createListingButton).toBeUndefined();
        });
        it('should show Create Listing button when the user is a member', () => {
            // given
            const props = {
                setAccountType,
                setIsAdmin,
                setAuthenticated,
                setActive,
                isAdmin: false,
                accountType: 'member',
                authenticated: true
            };

            // when
            const testInstance = renderer.create(<BrowserRouter><Header {...props}/></BrowserRouter>).root;
            const createListingButton = testInstance.findAllByType(Link)
                .find(element => element.props.children === 'Create Listing');

            // then
            expect(createListingButton).toBeDefined();
        });
        it('should show Create Listing button when the user is a business', () => {
            // given
            const props = {
                setAccountType,
                setIsAdmin,
                setAuthenticated,
                setActive,
                isAdmin: false,
                accountType: 'business',
                authenticated: true
            };

            // when
            const testInstance = renderer.create(<BrowserRouter><Header {...props}/></BrowserRouter>).root;
            const createListingButton = testInstance.findAllByType(Link)
                .find(element => element.props.children === 'Create Listing');

            // then
            expect(createListingButton).toBeDefined();
        });
    });

    describe('Admin button', () => {
        it('should show not show Admin button when the user is unauthenticated', () => {
            // given
            const props = {
                setAccountType,
                setIsAdmin,
                setAuthenticated,
                setActive,
                isAdmin: false,
                accountType: null,
                authenticated: false
            };

            // when
            const testInstance = renderer.create(<BrowserRouter><Header {...props}/></BrowserRouter>).root;
            const adminButton = testInstance.findAllByType(Link)
                .find(element => element.props.children === 'Admin');

            // then
            expect(adminButton).toBeUndefined();
        });
        it('should show not Admin button when isAdmin is false', () => {
            // given
            const props = {
                setAccountType,
                setIsAdmin,
                setAuthenticated,
                setActive,
                isAdmin: false,
                accountType: 'member',
                authenticated: true
            };

            // when
            const testInstance = renderer.create(<BrowserRouter><Header {...props}/></BrowserRouter>).root;
            const adminButton = testInstance.findAllByType(Link)
                .find(element => element.props.children === 'Admin');

            // then
            expect(adminButton).toBeUndefined();
        });
        it('should show Admin button when the user is authenticated and isAdmin is true', () => {
            // given
            const props = {
                setAccountType,
                setIsAdmin,
                setAuthenticated,
                setActive,
                isAdmin: true,
                accountType: 'member',
                authenticated: true
            };

            // when
            const testInstance = renderer.create(<BrowserRouter><Header {...props}/></BrowserRouter>).root;
            const adminButton = testInstance.findAllByType(Link)
                .find(element => element.props.children === 'Admin');

            // then
            expect(adminButton).toBeDefined();
        });
    });

    describe('Login button', () => {
        it('should show Login button when the user is unauthenticated', () => {
            // given
            const props = {
                setAccountType,
                setIsAdmin,
                setAuthenticated,
                setActive,
                isAdmin: false,
                accountType: null,
                authenticated: false
            };

            // when
            const testInstance = renderer.create(<BrowserRouter><Header {...props}/></BrowserRouter>).root;
            const loginButton = testInstance.findAllByType(Link)
                .find(element => element.props.children === 'Login');

            // then
            expect(loginButton).toBeDefined();
        });
        it('should not show Login button when user is authenticated', () => {
            // given
            const props = {
                setAccountType,
                setIsAdmin,
                setAuthenticated,
                setActive,
                isAdmin: false,
                accountType: 'member',
                authenticated: true
            };

            // when
            const testInstance = renderer.create(<BrowserRouter><Header {...props}/></BrowserRouter>).root;
            const loginButton = testInstance.findAllByType(Link)
                .find(element => element.props.children === 'Login');

            // then
            expect(loginButton).toBeUndefined();
        });
    });

    describe('Sign Up button', () => {
        it('should show Sign Up button when the user is unauthenticated', () => {
            // given
            const props = {
                setAccountType,
                setIsAdmin,
                setAuthenticated,
                setActive,
                isAdmin: false,
                accountType: null,
                authenticated: false
            };

            // when
            const testInstance = renderer.create(<BrowserRouter><Header {...props}/></BrowserRouter>).root;
            const signupButton = testInstance.findAllByType(Link)
                .find(element => element.props.children === 'Sign Up');

            // then
            expect(signupButton).toBeDefined();
        });
        it('should not show Sign Up button when user is authenticated', () => {
            // given
            const props = {
                setAccountType,
                setIsAdmin,
                setAuthenticated,
                setActive,
                isAdmin: false,
                accountType: 'member',
                authenticated: true
            };

            // when
            const testInstance = renderer.create(<BrowserRouter><Header {...props}/></BrowserRouter>).root;
            const signupButton = testInstance.findAllByType(Link)
                .find(element => element.props.children === 'Sign Up');

            // then
            expect(signupButton).toBeUndefined();
        });
    });

    describe('Logout button', () => {
        it('should not show Logout button when the user is unauthenticated', () => {
            // given
            const props = {
                setAccountType,
                setIsAdmin,
                setAuthenticated,
                setActive,
                isAdmin: false,
                accountType: null,
                authenticated: false
            };

            // when
            const testInstance = renderer.create(<BrowserRouter><Header {...props}/></BrowserRouter>).root;
            const logoutButton = testInstance.findAllByType(Link)
                .find(element => element.props.children === 'Logout');

            // then
            expect(logoutButton).toBeUndefined();
        });
        it('should show Logout button when user is authenticated', () => {
            // given
            const props = {
                setAccountType,
                setIsAdmin,
                setAuthenticated,
                setActive,
                isAdmin: false,
                accountType: 'member',
                authenticated: true
            };

            // when
            const testInstance = renderer.create(<BrowserRouter><Header {...props}/></BrowserRouter>).root;
            const logoutButton = testInstance.findAllByType(Link)
                .find(element => element.props.children === 'Logout');

            // then
            expect(logoutButton).toBeUndefined();
        });
    });
});