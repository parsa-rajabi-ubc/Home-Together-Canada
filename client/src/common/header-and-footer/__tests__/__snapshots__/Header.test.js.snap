// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Header structure should match snapshot test 1`] = `
<div>
  <nav
    className="top-0 flex w-full bg-green-400 tex-black"
  >
    <div
      className="container flex items-center justify-between w-full py-2 mx-auto mt-0 whitespace-no-wrap"
    >
      <div
        className="flex flex-no-wrap items-center ml-4"
      >
        <a
          className="text-lg font-bold lg:text-xl"
          href="/"
          onClick={[Function]}
        >
          Home Together Canada
        </a>
      </div>
      <div
        className="flex-no-wrap hidden w-full p-4 mt-2 tex-black lg:flex lg:items-center lg:w-auto lg:block lg:mt-0 lg:bg-transparent lg:p-0"
      >
        <div
          className="items-center justify-end flex-1 mr-16 list-reset lg:flex"
        >
          <a
            className="nav-icon"
            href="/members"
            onClick={[Function]}
          >
            Search Member Profiles
          </a>
          <a
            className="nav-icon"
            href="/services"
            onClick={[Function]}
          >
            Home Sharing Services
          </a>
          <a
            className="nav-icon"
            href="/classifieds"
            onClick={[Function]}
          >
            Local Classifieds
          </a>
          <a
            className="nav-icon"
            href="/faq"
            onClick={[Function]}
          >
            FAQ
          </a>
        </div>
        <a
          className="nav-btn"
          href="/login"
          onClick={[Function]}
        >
          Login
        </a>
        <a
          className="nav-btn"
          href="/registration"
          onClick={[Function]}
        >
          Sign Up
        </a>
      </div>
    </div>
    <hr
      className="py-0 my-0 border-b border-gray-200 opacity-25"
    />
  </nav>
</div>
`;
